### PR TITLE
Add KeepAlive settings between client and proxy 

### PR DESCRIPTION
**Pull request by fum1h1to**
# Description
- add KeepAlive Settings
    - Add a configuration setting to enable/disable KeepAlive between the client and authorization-proxy.

- config.yaml
```
version: v2.0.0
server:
  disableKeepAlives: false
```

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Non-code changes (update documentation, pipeline, etc.)

### Flags

- [ ] Breaks backward compatibility
- [ ] Requires a documentation update
- [ ] Has untestable code

## Related issue/PR

**Delete this section if there are no issues or pull requests that relate to this pull request.**
- Fixes #_issue_
- Closes #_PR_

---

## Checklist

- [ ] Followed the guidelines in the CONTRIBUTING document
- [ ] Added prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]` in the PR title if necessary
- [ ] Tested and linted the code
- [ ] Commented the code
- [ ] Made corresponding changes to the documentation
- [ ] Passed all pipeline checking

## Checklist for maintainer

- Use `Squash and merge`
- Double-confirm the merge message has prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]`
- Delete the branch after merge

### CODE REVIEW

File: config/config_test.go [hunk 1]
```diff
@@ -77,10 +77,11 @@ func TestNew(t *testing.T) {
...
 				Server: Server{
-					Port:            8082,
-					Timeout:         "10s",
-					ShutdownTimeout: "10s",
-					ShutdownDelay:   "9s",
+					Port:              8082,
+					Timeout:           "10s",
+					ShutdownTimeout:   "10s",
+					ShutdownDelay:     "9s",
+					DisableKeepAlives: false,
```
Comment (y-myajima) [inline config/config_test.go:84]:
```suggestion
					DisableKeepAlives: true,
```
初期値のことを考えて、trueでテストした方が良いと思いました

File: test/data/example_config.yaml [hunk 2]
```diff
@@ -5,6 +5,7 @@ server:
   timeout: 10s
   shutdownTimeout: 10s
   shutdownDelay: 9s
+  disableKeepAlives: false
```
Comment (y-myajima) [inline test/data/example_config.yaml:8]:
```suggestion
  disableKeepAlives: true
```

File: config/config.go [hunk 3]
```diff
@@ -64,6 +64,9 @@ type Server struct {
 	// ShutdownDelay represents the delay duration between the health check server shutdown and the api server shutdown.
 	ShutdownDelay string `yaml:"shutdownDelay"`
 
+	// DisableKeepAlives represents whether HTTP keep-alive connections should be disabled.
```
[the remark this answers is not in the document]
Comment (fum1h1to) [reply config/config.go]: fix: [fa1ff08](https://github.com/AthenZ/authorization-proxy/pull/41/commits/fa1ff08cbdc4cdb7b1a465aa59680cf9198996ef)

File: config/config_test.go (hunk 1 above)
Comment (fum1h1to) [reply config/config_test.go]: fix: [fa1ff08](https://github.com/AthenZ/authorization-proxy/pull/41/commits/fa1ff08cbdc4cdb7b1a465aa59680cf9198996ef)

File: test/data/example_config.yaml (hunk 2 above)
Comment (fum1h1to) [reply test/data/example_config.yaml]: fix: [fa1ff08](https://github.com/AthenZ/authorization-proxy/pull/41/commits/fa1ff08cbdc4cdb7b1a465aa59680cf9198996ef)

**Review by y-myajima — APPROVED** [review]
LGTM

**Pull request merged**